### PR TITLE
[RA-50] Display all initiatives on power plant detail page

### PR DESCRIPTION
**Pull request by JeremyGibson**
* Adds ``PowerPlant.plant_initiatives`` ManyToManyField that is editable in the PowerPlant admin
* Updates power plant detail view to return a distinct set of Initiatives pulled from both the ``PowerPlant`` and ``PowerPlant``'s related ``Projects``
* Displays distinct initiatives on the detail page as a single list.

Additionally, the admin ``Initiative`` edit page was inefficiently rendering many related objects in the DOM, so this updates the form to make ``affiliated_organizations``, ``affiliated_people``, and ``documents`` render as select2 fields.

### CODE REVIEW

File: infrastructure/templates/infrastructure/powerplant_detail.html [hunk 1]
```diff
@@ -121,9 +123,20 @@ <h3>Estimated Output</h3>
       <div class="row double">
           <section>
               <h3>Fuel</h3>
-              {% for fuel in object.fuels.all %}
-                <div>{{ fuel }}</div>
-              {% endfor %}
+              <div class="row double">
```
Comment (copelco) [inline infrastructure/templates/infrastructure/powerplant_detail.html]: Since these changes are unrelated to the fuel changes, ideally this branch would have come off of ``mou-backend`` directly.

File: infrastructure/templates/infrastructure/powerplant_detail.html [hunk 2]
```diff
@@ -132,6 +145,23 @@ <h3>CO2 Emissions</h3>
       </div>
     </section>
 
+    <section class="related-power-data">
+        <h2>Associated Project Initiatives</h2>
+        
+        {% for project in object.project_set.published %}
+        <div class="row">
+            <section>
+                <h4>{{ project.name }}</h4>
```
Comment (copelco) [inline infrastructure/templates/infrastructure/powerplant_detail.html]: This prints all related project names even if there are no associated initiatives.

screen shot 2018-12-28 at 5 17 44 pm

File: infrastructure/templates/infrastructure/powerplant_detail.html [hunk 3]
```diff
@@ -132,6 +145,23 @@ <h3>CO2 Emissions</h3>
...
+                <ul class="clean">
+                {% for pi in project.initiatives.all %}
+                    <li><a href="{{ pi.get_absolute_url }}">{{pi.name}}</li>
```
Comment (copelco) [inline infrastructure/templates/infrastructure/powerplant_detail.html]:
```suggestion
                    <li><a href="{{ pi.get_absolute_url }}">{{ pi.name }}</li>
```

File: infrastructure/templates/infrastructure/powerplant_detail.html [hunk 4]
```diff
@@ -132,6 +145,23 @@ <h3>CO2 Emissions</h3>
       </div>
     </section>
 
+    <section class="related-power-data">
+        <h2>Associated Project Initiatives</h2>
+        
+        {% for project in object.project_set.published %}
```
Comment (copelco) [inline infrastructure/templates/infrastructure/powerplant_detail.html]: I think you could cut down on the query count a bit by performing a more specific select within the view. For example:

```python
projects_with_initiatives = self.object.project_set.exclude(initiatives__isnull=True)
context['project_initiatives'] = projects_with_initiatives.prefetch_related('initiatives')
```

Comment (copelco) [reply infrastructure/templates/infrastructure/powerplant_detail.html]: Actually, upon further reading of the ticket re _Duplicates should be supressed_ , this'll need to be improved first.

Comment (copelco) [reply infrastructure/templates/infrastructure/powerplant_detail.html]: Probably easiest to come from Initiative, maybe something like:

```python
Initiative.objects.filter(project__power_plant_id=self.object.pk,
                          project__published=True).distinct().order_by('project__pk')
```

File: infrastructure/templates/infrastructure/powerplant_detail.html (hunk 1 above)
Comment (JeremyGibson) [reply infrastructure/templates/infrastructure/powerplant_detail.html]: I think did that because `docker-compose` wasn't on the mou-backend. Should I rebase it?

File: infrastructure/templates/infrastructure/powerplant_detail.html (hunk 2 above)
Comment (JeremyGibson) [reply infrastructure/templates/infrastructure/powerplant_detail.html]: Should handle this.
```
{% if project.initiatives.all %}
```

File: infrastructure/templates/infrastructure/powerplant_detail.html (hunk 4 above)
Comment (JeremyGibson) [reply infrastructure/templates/infrastructure/powerplant_detail.html]: The `distinct` here doesn't seem to work for me. I've set up a situation where i have a powerplant that assigned to 3 different projects and those projects have initiatives. Two of the projects have the "Karakoram Highway" as an Initiative.  I get this back:

```
Out[22]: <TreeQuerySet [<Initiative: Afghanistan National Development Strategy>, <Initiative: Karakoram Highway>, <Initiative: South Asia Subregional Economic Cooperation (SASEC)>, <Initiative: Belt and Road>, <Initiative: China-Pakistan Economic Corridor (CPEC)>, <Initiative: Karakoram Highway>]>
```

The query works if we leave out the `order_by`, but then we run into the same complications with grouping and display.

Comment (JeremyGibson) [reply infrastructure/templates/infrastructure/powerplant_detail.html]: The below commits are a bit solve this issue by choosing not to identify the project that the related initiative is attached to. Instead we just list "plant initiatives" and "associated project initiatives." Moved these list up to the top of the page.
